src/components/avatar-creator-raw.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { FC, useEffect, useRef } from 'react';
+import { AvatarCreatorConfig, IFrameEvent } from '../types';
+import { JSONTryParse } from '../utils';
+import { AvatarCreatorEvent } from '../events';
+import { useAvatarCreatorUrl } from '../hooks/use-avatar-creator-url';
+
+const MESSAGE_EVENT = 'message';
+const RPM_TARGET = 'readyplayerme';
+const IFRAME_READY_EVENT = 'v1.frame.ready';
+
+export type AvatarCreatorRawProps = {
+  subdomain: string;
+  className?: string;
+  config?: AvatarCreatorConfig;
+};
+
+export type EventReceivedProps = {
+  onEventReceived?: (event: IFrameEvent<AvatarCreatorEvent>) => void;
+};
+
+/**
+ * AvatarCreatorRaw is a React component that allows you to create an avatar using Ready Player Me and receive avatar URL. It exposes the raw events in one callback to allow you to write more custom logic around the event handling.
+ * @param subdomain The subdomain of your Ready Player Me instance.
+ * @param className The css styles to apply to this iframe.
+ * @param avatarCreatorConfig The configuration for the AvatarCreator component.
+ * @param onEventReceived A callback that is called when an avatar creator event is received.
+ * @returns A React component.
+ */
+export const AvatarCreatorRaw: FC<AvatarCreatorRawProps & EventReceivedProps> = ({ subdomain, className, config, onEventReceived }) => {
+  const iframeRef = useRef<HTMLIFrameElement | null>(null);
+  const url = useAvatarCreatorUrl(subdomain, config);
+
+  const subscribeToAvatarCreatorEvents = () => {
+    if (!iframeRef.current?.contentWindow) return;
```

Comment:
Is this needed here?

Reply:
Yeah, without it we get type errors.

![image](https://github.com/readyplayerme/react-avatar-creator/assets/121796327/027ddfae-2f73-485a-8941-515c1349dda2)